subsys/net/l2/ethernet/gptp/gptp_messages.c

```diff
@@ -538,25 +539,12 @@ struct net_pkt *gptp_prepare_announce(int port)
 	ann->cur_utc_offset = htons(global_ds->current_utc_offset);
 	ann->time_source = global_ds->time_source;
 
+	gm_prio = &global_ds->gm_priority;
 	switch (GPTP_PORT_BMCA_DATA(port)->info_is) {
 	case GPTP_INFO_IS_MINE:
-		ann->root_system_id.grand_master_prio1 = default_ds->priority1;
-		ann->root_system_id.grand_master_prio2 = default_ds->priority2;
-
-		ann->root_system_id.clk_quality.clock_accuracy =
-			default_ds->clk_quality.clock_accuracy;
-		ann->root_system_id.clk_quality.clock_class = default_ds->clk_quality.clock_class;
-		ann->root_system_id.clk_quality.offset_scaled_log_var =
-			htons(default_ds->clk_quality.offset_scaled_log_var);
-
-		memcpy(&ann->root_system_id.grand_master_id,
-		       default_ds->clk_id,
-		       GPTP_CLOCK_ID_LEN);
-		break;
 	case GPTP_INFO_IS_RECEIVED:
 		memcpy(&ann->root_system_id,
-		       &GPTP_PORT_BMCA_DATA(port)->
-				master_priority.root_system_id,
```

Comment:
@mschappa, thanks, the change looks fine to me.

I'm trying to figure out why DevB is not forwarding the announce message properly at the moment. Shouldn't master_priority.root_system_id already be same as gm_prio->root_system_id at this point? 
Or is it getting stuck in the GPTP_INFO_IS_MINE case?

Would you mind uploading a wireshark capture before and after the change, please?

Reply:
Hi @talih0, thanks for taking a look. At the moment I do not have access to the board but I can do a capture during next week probably. While debugging, for me it looked like it is always considering GPTP_INFO_IS_MINE case before sending announce messages and therefore, the information always contained the configured values instead the once received from the GM.

Reply:
@mschappa any progress on this?

Reply:
@jukkar @talih0 sorry for the delay. I was out of office until today.

Please find attached two wireshark logs _announce_before_fix.pcap_ and _announce_after_fix.pcap_ within [announce_pcaps.zip](https://github.com/zephyrproject-rtos/zephyr/files/13888173/announce_pcaps.zip). Both have been captured at DEV-C according to the issue description #66646:
 - Dev-C is identified as LCFCHeFe_7c:a9:ac
 - Dev-B is identified as RenesasE_00:00:01

When looking a the announce messages before the fix, we can see that DEV-B forwards its own priority1 as 200. After the fix it forwards the GM prioity1 as 100.

![grafik](https://github.com/zephyrproject-rtos/zephyr/assets/11267054/64383b66-4fef-4276-9934-2756e0bb06e3)
![grafik](https://github.com/zephyrproject-rtos/zephyr/assets/11267054/87c6b2a0-6c7e-4c76-96f2-4827fd1e3732)

Reply:
thanks, @mschappa, I'll look over tomorrow.